luminaire/exploration/data_exploration.py

```diff
@@ -798,14 +801,19 @@ def kf_naive_outlier_detection(self, input_series, idx_position):
         from pykalman import KalmanFilter
 
         kf = KalmanFilter()
-        filtered_state_means, filtered_state_covariance = kf.em(input_series).filter(input_series)
 
-        residuals = input_series - filtered_state_means[:,0]
+        truncated_series = input_series[-(self.min_ts_length * 3):] \
+            if idx_position == len(input_series) - 1 else input_series
+        idx_position = len(truncated_series) - 1
+
+        filtered_state_means, filtered_state_covariance = kf.em(truncated_series).filter(truncated_series)
+
+        residuals = truncated_series - filtered_state_means[:, 0]
 
         is_anomaly = residuals[idx_position] < np.mean(residuals) \
-                     - (2.5 * np.sqrt(filtered_state_covariance)[idx_position][0][0]) \
+                     - (1 * np.sqrt(filtered_state_covariance)[idx_position][0][0]) \
```

Comment:
Nit: Any reason for updating this from 2.5 to 1? Can we add some comments in the code.

Reply:
Comments added!

Reply:
The comment does not talk about those multiplier 2.5 changes to 1. Is that something anyone can easily understand? or you see anyone would be interested to know?

Reply:
The marginal word has been added to emphasize the shrink here.